js/localization/ldml/date.parser.js

```diff
@@ -218,6 +219,22 @@ export const getRegExpInfo = function(format, dateParts) {
 
     addPreviousStub();
 
+    patterns.every((pattern, index) => {
+        const char = pattern[0];
+        const prevPatternChar = index ? patterns[index - 1][0] : null;
+        const nextPatternChar = index < patterns.length - 1 ? patterns[index + 1][0] : null;
+        const isSingle = pattern.length === 1;
+        const isFirst = index === 0;
+        const isRegexpPart = (char) => PATTERN_REGEXPS[char] && char !== ':';
+
+        if(isSingle && isRegexpPart(char) && (isRegexpPart(prevPatternChar) || (isFirst && isRegexpPart(nextPatternChar)))) {
```

Comment:
I prefer to extract such long conditions to the separate variable. It's name should make the code to be easier to read.